File: c programming/st.c
// implement stack an an array
#include <stdio.h>
#include <stdlib.h>
#define MAX 100

int stack[MAX];
int top = -1;

void push(int data)
{
    if (top == MAX - 1)
    {
        printf("Stack Overflow\n");
    }
    else
    {
        stack[++top] = data;
    }
}

void pop()
{
    if (top == -1)
    {
        printf("STack Underflow\n");
    }
    else
    {
        printf("Popped element: %d\n", stack[top--]);
    }
}

void peek()
{
    if (top == -1)
    {
        printf("Stack is empty\n");
    }
    else
    {
        printf("Top element: %d\n", stack[top]);
    }
}
void display()
{
    if (top == -1)
    {
        printf("Stack is empty\n");
    }
    else
    {
        for (int i = top; i >= 0; i--)
        {
            printf("%d\n", stack[i]);
        }
    }
}
int main()
{
    int choice, data;
    while (1)
    {
        printf("1. Push\n");
        printf("2. Pop\n");
        printf("3. Peek\n");
        printf("4. Display\n");
        printf("5. Exit\n");
        printf("Enter your choice: ");
        scanf("%d", &choice);
        switch (choice)
        {
        case 1:
            printf("Enter data to push: ");
            scanf("%d", &data);
            push(data);
            break;
        case 2:
            pop();
            break;
        case 3:
            peek();
            break;
        case 4:
            display();
            break;
        case 5:
            exit(0);
        default:
            printf("Invalid choice\n");
        }
    }
    return 0;
}

File: st.java
// implement a stack using array
public class st {
    int stack[] = new int[5];
    int top = -1;

    public void push(int data) {
        if (top == 4) {
            System.out.println("stack overflow");
        } else {
            top++;
            stack[top] = data;
        }
    }

    public void pop() {
        if (top == -1) {
            System.out.println("stack underflow");
        } else {
            top--;
        }
    }

    public void display() {
        for (int i = top; i >= 0; i--) {
            System.out.println("Element at index " + i + " is " + stack[i]);
        }
    }

    public void peek() {
        System.out.println(stack[top]);
    }

    public void isEmpty() {
        if (top == -1) {
            System.out.println("stack is empty");
        } else {
            System.out.println("stack is not empty");
        }
    }

    public static void main(String[] args) {
        st s = new st();
        s.push(1);
        s.push(2);
        s.push(3);
        s.push(4);
        s.push(5);
        s.display();
        s.pop();
        s.display();
        s.peek();
        s.display();
        s.isEmpty();
        s.display();
    }
}